cowamm/cow_amm_cow_with_users_surplus_query_4031724.sql

```diff
@@ -1,56 +1,78 @@
-            else token_2_transfer_usd + (token_2_balance_usd - token_2_transfer_usd) * token_1_transfer_usd / token_1_balance_usd
-        end as surplus
-    from dune.cowprotocol.result_balancer_cow_amm_base_query_v_2
-    where istrade
+        'ethereum' as blockchain,
+        cow_per_batch.block_time,
+        cow_per_batch.tx_hash,
+        solvers.name as solver_name,
+        naive_cow,  -- fraction of batch volume traded within a CoW
+        trades.surplus_usd as surplus_in_usd,  -- surplus of the executed CoW AMM order, expressed in USD
+        naive_cow * trades.surplus_usd as realized_cow_surplus_in_usd -- surplus of the CoW AMM that is assumed to be generated via a CoW.
+    from "query_4025739(blockchain='ethereum',start_time='{{start_time}}',end_time='{{end_time}}')" as cow_per_batch
+    inner join cow_protocol_ethereum.trades as trades on cow_per_batch.tx_hash = trades.tx_hash
+    inner join cow_protocol_ethereum.batches as batches on cow_per_batch.tx_hash = batches.tx_hash
+    inner join cow_protocol_ethereum.solvers as solvers on batches.solver_address = solvers.address and solvers.active
+    where trades.trader in (select address from query_3959044 where blockchain = 'ethereum')
 ),
 
-cow_surplus_per_batch as (
+cow_surplus_per_batch_gnosis as (
     select
+        'gnosis' as blockchain,
         cow_per_batch.block_time,
         cow_per_batch.tx_hash,
-        solver_address,
+        solvers.name as solver_name,
         naive_cow,  -- fraction of batch volume traded within a CoW
-        surplus as surplus_in_usd,  -- surplus of the executed CoW AMM order, expressed in USD
-        naive_cow * surplus as realized_cow_surplus_in_usd -- surplus of the CoW AMM that is assumed to be generated via a CoW.
-    from "query_4025739(blockchain='{{blockchain}}',start_time='{{start_time}}',end_time='{{end_time}}')" as cow_per_batch
-    inner join cow_amm_surplus on cow_per_batch.tx_hash = cow_amm_surplus.tx_hash
-    inner join cow_protocol_{{blockchain}}.batches as b on cow_per_batch.tx_hash = b.tx_hash
+        trades.surplus_usd as surplus_in_usd,  -- surplus of the executed CoW AMM order, expressed in USD
+        naive_cow * trades.surplus_usd as realized_cow_surplus_in_usd -- surplus of the CoW AMM that is assumed to be generated via a CoW.
+    from "query_4025739(blockchain='gnosis',start_time='{{start_time}}',end_time='{{end_time}}')" as cow_per_batch
+    inner join cow_protocol_gnosis.trades as trades on cow_per_batch.tx_hash = trades.tx_hash
+    inner join cow_protocol_gnosis.batches as batches on cow_per_batch.tx_hash = batches.tx_hash
+    inner join cow_protocol_gnosis.solvers as solvers on batches.solver_address = solvers.address and solvers.active
+    where trades.trader in (select address from query_3959044 where blockchain = 'gnosis')
+),
+
+cow_surplus_per_batch as (
```

Comment:
Why are we not including arbitrum AMMs here? I think they also qualify for the bounty.

Reply:
I added a table for arbitrum. Interestingly, there is no difference in total surplus. I briefly checked the intermediate tables and they seem to work. So I think it is fine to merge this.